include/pybind11/stl/filesystem.h

```diff
@@ -13,21 +13,30 @@
 #include <string>
 
 #ifdef __has_include
-#    if defined(PYBIND11_CPP17) && __has_include(<filesystem>)
-#        include <filesystem>
-#        define PYBIND11_HAS_FILESYSTEM 1
+#    if defined(PYBIND11_CPP17)
+#        if __has_include(<filesystem>) && \
+          PY_VERSION_HEX >= 0x03060000
+#            include <filesystem>
+#            define PYBIND11_HAS_FILESYSTEM 1
+namespace fs = std::filesystem;
+#        elif __has_include(<experimental/filesystem>)
+#            include <experimental/filesystem>
+#            define PYBIND11_HAS_EXPERIMENTAL_FILESYSTEM
```

Comment:
Slightly silly, but for local consistency, could you also make this
```
define PYBIND11_HAS_EXPERIMENTAL_FILESYSTEM 1
```
like the existing define?